pandas/tests/test_base.py

```diff
@@ -187,7 +187,24 @@ def setup_method(self, method):
         types = ["bool", "int", "float", "dt", "dt_tz", "period", "string", "unicode"]
         self.indexes = [getattr(self, "{}_index".format(t)) for t in types]
         self.series = [getattr(self, "{}_series".format(t)) for t in types]
-        self.objs = self.indexes + self.series
+
+        # To test narrow dtypes, we use narrower *data* elements, not *index* elements
```

Comment:
this whole thing badly needs parameterization.

can you pull out the unique tests and do that instead of repeating all of this?